### ISSUE TITLE
New debug packages on Debian Stretch

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#2734]
see https://wiki.debian.org/AutomaticDebugPackages - they are now in their own repo. We should probably adjust the stacktrace instructions.

**Comment by tulikavijay** [comment]
Hello! I am new to open source and i would like to contribute. I have read the contribution guidelines and i would like to start. If only you could guide me where to make changes or anything else really :)

**Comment by The-Compiler** [comment]
Hey @tulikavijay! As mentioned in the [contribution docs](https://github.com/qutebrowser/qutebrowser/blob/master/CONTRIBUTING.asciidoc), I'm currently very busy with exams coming up, so I'd prefer for contributions to be postponed until September.

However, I suggest you start using qutebrowser if you haven't yet, and then join the [IRC channel](https://github.com/qutebrowser/qutebrowser#getting-help) to get to know the community - the channel is also a good way to get fast answers to questions you might have.

**Comment by tulikavijay** [comment]
I will! thanks for the reply.

**Comment by xtaran** [comment]
Relevant place which needs an update: https://github.com/qutebrowser/qutebrowser/blob/master/doc/stacktrace.asciidoc#getting-debugging-symbols

**Comment by xtaran** [comment]
Not for Debian Stretch (at least not with the list of packages in that command), but for Debian Buster at least: `libqt5webkit5-dbg` has been replaced with `libqt5webkit5-dbgsym` and that one comes from the extra debug symbols package.

I currently don't know by mind how this works with Ubuntu. AFAIK their implementation of automatic debug symbols is older than Debian's and slightly different. (IIRC they call or at least called them `ddeb`s.)

But the general question here is: How to handle (documentation-wise) the fact that the debug package name for any of these packages may change over time and not generally from one stable release to another?

**Comment by The-Compiler** [comment]
Oh, I just realized QtWebEngine isn't even in that list, that should probably be added as well. I can't find any `-dbg` package for that though.

How do I find out more about those? On packages.debian.org as well as tracker.debian.org I can't find anything about debug packages at all.

> I currently don't know by mind how this works with Ubuntu. AFAIK their implementation of automatic debug symbols is older than Debian's and slightly different. (IIRC they call or at least called them ddebs.)

FWIW at least on Ubuntu 16.04, 17.10 and 18.04 that command seems to work fine. On 18.10, it doesn't find `libqt5webkit5-dbg`, probably for the same reason than on Buster.

> But the general question here is: How to handle (documentation-wise) the fact that the debug package name for any of these packages may change over time and not generally from one stable release to another?

Even if it's a bit of effort, I'd like to have a list of packages to install for every case, similar to the [install docs](https://github.com/qutebrowser/qutebrowser/blob/master/doc/install.asciidoc#on-debian--ubuntu). It's a one-time effort (compared to people finding out what to do by themselves), and it makes it easier for people to provide good stack traces, which hopefully means more people will do so.

**Comment by xtaran** [comment]
On Stretch (with repo for automatic debug symbols enabled):

```
~ → apt-cache search -n 'qt5?webengine.*-dbg'
python-pyqt5.qtwebengine-dbg - Python 2 bindings for Qt5's WebEngine module (debug extensions)
python3-pyqt5.qtwebengine-dbg - Python 3 bindings for Qt5's WebEngine module (debug extensions)
libqt5webengine5-dbgsym - Debug symbols for libqt5webengine5
libqt5webenginecore5-dbgsym - Debug symbols for libqt5webenginecore5
libqt5webenginewidgets5-dbgsym - Debug symbols for libqt5webenginewidgets5
qml-module-qtwebengine-dbgsym - Debug symbols for qml-module-qtwebengine
qtwebengine5-examples-dbgsym - Debug symbols for qtwebengine5-examples
```

**Comment by xtaran** [comment]
What might help on future Debian releases (yet untested on Ubuntu): Since version 0.79 (November 2017), [debian-goodies](https://packages.debian.org/debian-goodies) contains a tool named `find-dbgsym-packages` ([man page](https://manpages.debian.org/testing/debian-goodies/find-dbgsym-packages.1.en.html)).

**Comment by The-Compiler** [comment]
Fixed (I think) via 1642fa252ae9427eef3acdba3ce45782bad608e2
